src/dotnet-roslyn-tools/CreateReleaseTags/CreateReleaseTags.cs

```diff
@@ -2,42 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the License.txt file in the project root for more information.
 
-using System.Collections.Immutable;
-using System.Net;
-using System.Text.RegularExpressions;
-using System.Xml.Linq;
 using LibGit2Sharp;
 using Microsoft.Extensions.Logging;
 using Microsoft.RoslynTools.Authentication;
 using Microsoft.RoslynTools.Products;
 using Microsoft.RoslynTools.Utilities;
+using Microsoft.RoslynTools.VS;
 using Microsoft.TeamFoundation.SourceControl.WebApi;
 using Microsoft.VisualStudio.Services.Common;
 using Newtonsoft.Json.Linq;
+using System.Collections.Immutable;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
 
 namespace Microsoft.RoslynTools.CreateReleaseTags;
 
 internal static class CreateReleaseTags
 {
-    private static Roslyn s_roslynInfo = new();
-    public static async Task<int> CreateReleaseTagsAsync(RoslynToolsSettings settings, ILogger logger)
+    public static async Task<int> CreateReleaseTagsAsync(string productName, RoslynToolsSettings settings, ILogger logger)
     {
         try
         {
             using var devdivConnection = new AzDOConnection(settings.DevDivAzureDevOpsBaseUri, "DevDiv", settings.DevDivAzureDevOpsToken);
             using var dncengConnection = new AzDOConnection(settings.DncEngAzureDevOpsBaseUri, "internal", settings.DncEngAzureDevOpsToken);
 
-            var connections = new[] { devdivConnection, dncengConnection };
+            var product = VSBranchInfo.AllProducts.Single(p => p.Name.Equals(productName, StringComparison.OrdinalIgnoreCase));
```

Comment:
Should this mapping be happening in the command handler so this way this method just directly is given the IProduct?

Reply:
It could, but the command handlers in general have no logic, so this makes more sense to me. It would be a literal move for no benefit IMO

Reply:
To me this is the prefect logic that should go in the command handler: it's parsing the input into the workable input. The other two parameters are already rich types, after all.

Reply:
You're not wrong. Whats _really_ annoying is that System.CommandLine is already doing the same parsing, and validating that the input is correct and in the expected set. I'll see if there is a better API for getting strongly typed arguments out of it